### PR TITLE
Update to latest Jekyll format and add CI config

### PR DESCRIPTION
**Pull request by nschonni**
1. This adds Travis CI checking for the validity of links when stuff is PR'd.
1. Updates to use the structure and gem setup used by the GitHub sites

__
There are still broken links that will fail the build right now, but I believe they are being looked at in other threads so they were left out of this
```
- ./_site/posts/how-to-hide-content/index.html
  *  External link https://developer.yahoo.com/blogs/ydn/clip-hidden-content-better-accessibility-53456.html failed: 404 No error
- ./_site/posts/understanding-visual-impairment/index.html
  *  External link http://www.ffb.ca/eye_conditions/RD_diseases.html failed: 404 No error
- ./_site/posts/what-is-color-contrast/index.html
  *  External link https://developer.yahoo.com/blogs/ydn/techniques-high-contrast-friendly-icons-153038779.html failed: 404 No error
- ./_site/resources.html
  *  External link http://www.accessifyforum.com/ failed: response code 0 means something's wrong.
             It's possible libcurl couldn't connect to the server or perhaps the request timed out.
             Sometimes, making too many requests at once also breaks things.
             Either way, the return message (if any) from the server is: Couldn't resolve host name
  *  External link http://yaccessibilityblog.com/ failed: 503 No error
  *  External link https://medium.com/web-accessibility/8f7161340cc1 failed: 404 No error
```

### CODE REVIEW

**Comment by davatron5000** [comment]
Whoah. This is great. What's required to get this setup and testing locally?

Also, I noticed a few files were deleted. Notably, `sitemap.xml` and `atom.xml`. Any reason for that?

**Comment by nschonni** [comment]
> Whoah. This is great. What's required to get this setup and testing locally?

Same as before, just `bundle install` and `bundle exec jekyll build`. For the validation of links it would be `bundle exec rake test`, but that could be added to a script like other repos use.

> Also, I noticed a few files were deleted. Notably, sitemap.xml and atom.xml. Any reason for that?

GitHub whitelisted the plugins to generate those files now automagically. It's all in separate commits, so the can be split out if preferred.

**Comment by nschonni** [comment]
Oh, and you'll need to turn on the Travis-CI hook to get PRs tested

**Comment by nschonni** [comment]
@davatron5000 let me know when you're done the resources triage, then I'll rebase this.

**Comment by davatron5000** [comment]
@nschonni We should be in great shape. I knocked all but #411 (the open ended one out).

And just for reference, since you're using the `github-pages` gem, this would close #440 as well.

**Comment by nschonni** [comment]
Rebased, and I've left the 3 failing urls for now https://travis-ci.org/nschonni/a11yproject.com/builds/197441998

**Comment by joe-watkins** [comment]
@nschonni Can you look into the merge conflicts here?

File: patterns.html [hunk 1]
```diff
@@ -37,22 +38,15 @@ <h3 class="article-section__title">{{ category.pattern-category }}</h3>
             </ul>
           </li>
         </ul>
-        {% for pattern in category.patterns | sort: 'title' %}
```
Comment (nschonni) [inline patterns.html]: Since the sorting method was causing an error, I deleted the duplicate block and fixed it once

**Comment by nschonni** [comment]
@joe-watkins let me know when you want to look at this, i just don't want to keep rebasing it

**Comment by joe-watkins** [comment]
@nschonni Things are looking good on my end when I pull it down and run things. I'm not sure about the Travis-CI hook. Is the thought to check the site for broken links and if some are found hold back the build?

**Comment by nschonni** [comment]
Yes, the build should fail if there are broken links on the site.
For configuring the Travis hook, you can follow Steps 2 & 3 from https://docs.travis-ci.com/user/for-beginners
I could also add in some of the boilerplate scripts that GH uses from https://github.com/github/scripts-to-rule-them-all

**Comment by joe-watkins** [comment]
@nschonni Tnx for the links. I've never messed w/Travis before so links are helpful.

I like the idea but concerned about results. For instance when I run `bundle exec rake test` I get these results:

```
Running ["ScriptCheck", "LinkCheck", "ImageCheck", "HtmlCheck"] on ["./_site/"] on *.html...


Checking 255 external links...
Ran on 29 files!


- ./_site/about.html
  *  External link http://www.census.gov/main/www/popclock.html failed: 301 SSL connect error
- ./_site/posts/how-to-hide-content/index.html
  *  External link https://developer.yahoo.com/blogs/ydn/clip-hidden-content-better-accessibility-53456.html failed: 504 No error
- ./_site/posts/how-to-use-application-role/index.html
  *  External link https://dvcs.w3.org/hg/aria-unofficial/raw-file/tip/index.html#application failed: 503 No error
- ./_site/posts/skip-nav-links/index.html
  *  External link http://www.nczonline.net/blog/2013/01/15/fixing-skip-to-content-links failed: 301 SSL connect error
- ./_site/posts/understanding-visual-impairment/index.html
  *  External link http://www.ffb.ca/eye_conditions/RD_diseases.html failed: 404 No error
- ./_site/posts/what-is-color-contrast/index.html
  *  External link https://developer.yahoo.com/blogs/ydn/techniques-high-contrast-friendly-icons-153038779.html failed: 404 No error
- ./_site/resources.html
  *  External link http://allyjs.io failed: 301 SSL connect error
  *  External link http://bitsofco.de/2015/the-accessibility-cheatsheet failed: 301 SSL connect error
  *  External link http://quailjs.org failed: response code 0 means something's wrong.
             It's possible libcurl couldn't connect to the server or perhaps the request timed out.
             Sometimes, making too many requests at once also breaks things.
             Either way, the return message (if any) from the server is: Couldn't resolve host name
  *  External link http://rawgithub.com/w3c/aria-in-html/master/index.html failed: 301 SSL connect error
  *  External link https://frend.co failed: response code 0 means something's wrong.
             It's possible libcurl couldn't connect to the server or perhaps the request timed out.
             Sometimes, making too many requests at once also breaks things.
             Either way, the return message (if any) from the server is: SSL connect error
rake aborted!
HTML-Proofer found 11 failures!
/usr/local/lib/ruby/gems/2.3.0/gems/html-proofer-3.5.0/lib/html-proofer/runner.rb:171:in `print_failed_tests'
/usr/local/lib/ruby/gems/2.3.0/gems/html-proofer-3.5.0/lib/html-proofer/runner.rb:52:in `run'
/Users/joewatkins/Desktop/a11y-test/a11yproject.com/Rakefile:7:in `block in <top (required)>'
/usr/local/lib/ruby/gems/2.3.0/gems/rake-12.0.0/exe/rake:27:in `<top (required)>'
/usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.14.3/lib/bundler/cli/exec.rb:74:in `load'
/usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.14.3/lib/bundler/cli/exec.rb:74:in `kernel_load'
/usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.14.3/lib/bundler/cli/exec.rb:27:in `run'
/usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.14.3/lib/bundler/cli.rb:335:in `exec'
/usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.14.3/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
/usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.14.3/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
/usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.14.3/lib/bundler/vendor/thor/lib/thor.rb:359:in `dispatch'
/usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.14.3/lib/bundler/cli.rb:20:in `dispatch'
/usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.14.3/lib/bundler/vendor/thor/lib/thor/base.rb:440:in `start'
/usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.14.3/lib/bundler/cli.rb:11:in `start'
/usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.14.3/exe/bundle:32:in `block in <top (required)>'
/usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.14.3/lib/bundler/friendly_errors.rb:121:in `with_friendly_errors'
/usr/local/lib/ruby/gems/2.3.0/gems/bundler-1.14.3/exe/bundle:24:in `<top (required)>'
/usr/local/bin/bundle:22:in `load'
/usr/local/bin/bundle:22:in `<main>'
Tasks: TOP => test
(See full trace by running task with --trace)
```

Not sure if all the errors at the end are to be expected or not but more importantly getting 301 errors for sites that work e.g. http://allyjs.io

What ya think?

**Comment by nschonni** [comment]
Added some scripts and instructions along with the rebase. There are some failures that can either be dealt with now, or after this and fix up the build.

**Comment by nschonni** [comment]
Nevermind, I just fixed the links in separate commits https://travis-ci.org/nschonni/a11yproject.com/builds/211635963.
For some reason the population clock is failing on Travis, but not locally now

**Comment by nschonni** [comment]
Lol, another link is now dead after I restarted travis. Looks like http://www.dmoz.org is done, but a mirror has been setup at http://dmoztools.net/

**Comment by joe-watkins** [comment]
@nschonni Merged and Travis CI set up! And tested a build or two.. works great. Nice ASCII art btw :)

Can other members of the organization manage the Travis CI setup or just me?

**Comment by nschonni** [comment]
any repository admin can edit the hooks and reset builds. Anyone can also setup their own for their forks too.

**Pull request merged**